app/routes.py

```diff
@@ -0,0 +1,345 @@
+    # if request.method == "POST":
+    request_body = request.get_json()
+    customer_id = request_body.get("customer_id")
+    video_id = request_body.get("video_id")
+    due_date = date.today() + timedelta(days=7)
+
+    customer = Customer.query.get(customer_id)
+    if customer_id is None:
+        return jsonify(None), 400
+    if customer is None:
+        return jsonify(None), 404
+
+    video = Video.query.get(video_id)
+    if video_id is None:
+        return jsonify(None), 400
+    if video is None:
+        return jsonify(None), 404
+    
+
+    video_checked_out_count = Rental.query.filter_by(video_id=video_id, checked_in_status=False).count()
+    available_inventory = video.total_inventory - video_checked_out_count
+
+    if available_inventory == 0:
+        return jsonify({
+            "message" : "Could not perform checkout"
+        }), 400
+
+    new_rental = Rental(
+        customer_id=customer.id,
+        video_id=video.id,
+        due_date=due_date
+    )
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    video_checked_out_count = Rental.query.filter_by(video_id=video_id).count()
+    available_inventory = video.total_inventory - video_checked_out_count
+
+    return {
+        "customer_id": new_rental.customer_id,
+        "video_id": new_rental.video_id,
+        "due_date": new_rental.due_date,
+        "videos_checked_out_count": video_checked_out_count,
+        "available_inventory": available_inventory 
+        }, 200
+ 
+@rentals_bp.route("/check-in", methods=["POST"])
+def handle_rentals_in():
+    request_body = request.get_json()
+    customer_id = request_body.get("customer_id")
+    video_id = request_body.get("video_id")
+
+    customer = Customer.query.get(customer_id)
+    if customer_id is None:
+        return jsonify(None), 400
+    if customer is None:
+        return jsonify(None), 404
+
+    video = Video.query.get(video_id)
+    if video_id is None:
+        return jsonify(None), 400
+    if video is None:
+        return jsonify(None), 404
+
+    rental = Rental.query.filter_by(customer_id=customer_id, video_id=video_id, checked_in_status=False).order_by(Rental.due_date.asc()).first()
+    if rental is None:
+        return jsonify({"message" : f"No outstanding rentals for customer {customer_id} and video {video_id}"
+        }), 400
+
+    rental.checked_in_status = True
+    video_checked_out_count = Rental.query.filter_by(video_id=video_id, checked_in_status=False).count()
+    available_inventory = video.total_inventory - video_checked_out_count
+
+    db.session.commit()
+
+    return {
+        "customer_id": rental.customer_id,
+        "video_id": rental.video_id,
+        "videos_checked_out_count": video_checked_out_count,
+        "available_inventory": available_inventory 
+        }, 200
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def list_customer_rentals(customer_id):
+    customer = Customer.query.get(customer_id)
```

Comment:
We should have similar validation for the `customer_id` as we have for the CRUD customer endpoints (no non-numeric values allowed).